ArduCopter/GCS_Copter.cpp

```diff
@@ -183,3 +183,24 @@ void GCS_Copter::update_vehicle_sensor_status_flags(void)
     }
 #endif
 }
+
+bool GCS_Copter::create_gcs_mavlink_backend(AP_SerialManager &serial_manager,
```

Comment:
I'm unhappy at cloning this method for each vehicle.

I *think* I could probably store the pointers in the base class - but I'd then have to cast them back into their true (subclass) pointer type in the `chan(uint8_t)` method.  If I were to do that this method would just return the `new`'d object.

Does that seem reasonable to anybody?

Reply:
I've factored this and moved it up, and cast the pointer on the getter.